sdk/eventhubs/azure-eventhubs/src/main/java/com/azure/messaging/eventhubs/BatchOptions.java

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.eventhubs;
+
+/**
+ * The set of options that can be specified when creating an {@link EventDataBatch}.
+ *
+ * @see EventHubProducer#createBatch()
+ * @see EventHubProducer#createBatch(BatchOptions)
+ */
+public class BatchOptions implements Cloneable {
+    private int maximumSizeInBytes;
+    private String partitionKey;
+
+    /**
+     * Sets the maximum size for the {@link EventDataBatch batch of events}, in bytes.
+     *
+     * @param maximumSizeInBytes The maximum size to allow for the {@link EventDataBatch batch of events}.
+     * @return The updated {@link BatchOptions} object.
+     */
+    public BatchOptions maximumSizeInBytes(int maximumSizeInBytes) {
+        this.maximumSizeInBytes = maximumSizeInBytes;
+        return this;
+    }
+
+    /**
+     * Gets the maximum size to allow for the batch of events, in bytes.
+     *
+     * @return The maximum size to allow for a single batch of events, in bytes.
+     */
+    public int maximumSizeInBytes() {
+        return maximumSizeInBytes;
+    }
+
+    /**
+     * Sets a hashing key to be provided for the batch of events, which instructs the Event Hubs service map this key to
+     * a specific partition but allowing the service to choose an arbitrary, partition for this batch of events and any
+     * other batches using the same partition hashing key.
+     *
+     * The selection of a partition is stable for a given partition hashing key. Should any other batches of events be
+     * sent using the same exact partition hashing key, the Event Hubs service will route them all to the same
+     * partition.
+     *
+     * This should be specified only when there is a need to group events by partition, but there is flexibility into
+     * which partition they are routed. If ensuring that a batch of events is sent only to a specific partition, it is
+     * recommended that the identifier of the position be specified directly when sending the batch.
```

Comment:
Same with this paragraph - I can feel my eyes crossing over. It might benefit from either simplification or sample code (or a link to sample code elsewhere).